### PR TITLE
chore: made a new component infoicontippy

### PR DESCRIPTION
**Pull request by randhir-devtron**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Common/CustomInput/CustomInput.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import React from 'react'
```
Comment (AbhishekA1509) [inline src/Common/CustomInput/CustomInput.tsx:1]: No need

File: src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        case 'MEDIUM':
+            className = TIPPY_ICON_DIM.MEDIUM
+            break
+        case 'LARGE':
+            className = TIPPY_ICON_DIM.LARGE
+            break
+        default:
+            className = TIPPY_ICON_DIM.MEDIUM
+    }
+    className += ' fcn-9 ml-8 cursor'
```
Comment (AbhishekA1509) [inline src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx:29]: Do not use margin, can use ```p-0 h-20 dc__no-background dc__no-border dc__outline-none-imp``` as baseClassname

File: src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+export const InfoIconWithTippy = ({
+    titleText,
+    infoText,
+    documentationLink,
+    additionalContent,
+    variant = TIPPY_ICON_DIM.LARGE,
+}: InfoIconTippyType) => {
+    let className: string
+    switch (variant) {
```
Comment (AbhishekA1509) [inline src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx:16]: Move this to util and avoid let

File: src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            className="w-300 h-100 fcv-5"
+            placement="right"
+            Icon={HelpIcon}
+            heading={titleText}
+            infoText={infoText}
+            showCloseButton
+            trigger="click"
+            interactive
+            documentationLink={documentationLink}
+            documentationLinkText="View Documentation"
```
Comment (AbhishekA1509) [inline src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx:42]: Are you sure we need to show documentationLinkText every time, we can move this to prop

File: src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+export const InfoIconWithTippy = ({
+    titleText,
+    infoText,
+    documentationLink,
+    additionalContent,
+    variant = TIPPY_ICON_DIM.LARGE,
+}: InfoIconTippyType) => {
+    let className: string
+    switch (variant) {
+        case 'SMALL':
```
Comment (AbhishekA1509) [inline src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx:17]: use enum

File: src/Shared/types.ts [hunk 6]
```diff
@@ -10,3 +12,11 @@ export enum CIMaterialSidebarType {
     CODE_SOURCE = 'Code Source',
     PARAMETERS = 'Parameters',
 }
+
+export interface InfoIconTippyType {
```
Comment (AbhishekA1509) [inline src/Shared/types.ts:16]: use Pick and import type from TippyCustomized

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/Common/AppStatus/utils.ts [hunk 7]
```diff
@@ -13,3 +13,22 @@ export const triggerStatus = (triggerDetailStatus: string): string => {
     }
     return triggerDetailStatus
 }
+
+export const getInfoIconTippyClass = (variant: string): string => {
+    let className: string
```
Comment (AbhishekA1509) [inline src/Common/AppStatus/utils.ts:18]: No need to create variable can directly return variable

File: src/Common/Constants.ts [hunk 8]
```diff
@@ -286,3 +286,8 @@ export enum ReactSelectInputAction {
 }
 
 export const ZERO_TIME_STRING = '0001-01-01T00:00:00Z'
+export enum TIPPY_ICON_DIM {
```
Comment (AbhishekA1509) [inline src/Common/Constants.ts:289]: Move it to types.ts and rename TippySizeVariant some thing like that

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/Common/AppStatus/AppStatus.tsx [hunk 9]
```diff
@@ -1,4 +1,5 @@
 import Tippy from '@tippyjs/react'
+import React from 'react'
```
Comment (AbhishekA1509) [inline src/Common/AppStatus/AppStatus.tsx:2]: Remove this

File: src/Common/AppStatus/utils.ts [hunk 10]
```diff
@@ -13,3 +12,4 @@ export const triggerStatus = (triggerDetailStatus: string): string => {
     }
     return triggerDetailStatus
 }
+
```
Comment (AbhishekA1509) [inline src/Common/AppStatus/utils.ts:15]: Run prettier once

File: src/Common/ImageTags.tsx [hunk 11]
```diff
@@ -257,21 +252,12 @@ export const ImageTagsContainer = ({
...
-            theme={TippyTheme.white}
-            className="w-300 fcv-5"
-            placement="right"
-            Icon={QuestionFilled}
+        <InfoIconTippy
             heading="Release tags"
-            showCloseButton
-            trigger="click"
-            interactive
+            infoText=""
```
Comment (AbhishekA1509) [inline src/Common/ImageTags.tsx:257]: no need to send this, this should be optional

File: src/Common/Constants.ts (hunk 8 above)
Comment (AbhishekA1509) [reply src/Common/Constants.ts:290]: Remove this if not using in any place

File: src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+        showCloseButton
+        trigger="click"
+        interactive
+        documentationLink={documentationLink}
+        documentationLinkText={documentationLinkText}
+        additionalContent={additionalContent}
+    >
+        <button
+            type="button"
+            className="p-0 dc__no-background dc__no-border dc__outline-none-imp flex ml-4 dc__tab-focus"
```
Comment (AbhishekA1509) [inline src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx:35]: Can we remove this ml-4?

File: src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+    heading,
+    infoText,
+    iconClass = 'fcv-5',
+    documentationLink,
+    documentationLinkText,
+    additionalContent,
+    className = 'icon-dim-16',
+    placement = 'bottom',
+    dataTestid,
+}: Pick<
```
Comment (AbhishekA1509) [inline src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx:16]: Move it to interface like export interface InfoIconTippyProps extends Pick< TippyCustomizedProps , 'heading' | ... > {dataTestid?: string}

File: src/Common/AppStatus/utils.ts [hunk 14]
```diff
@@ -1,5 +1,4 @@
 import { TIMELINE_STATUS } from './constants'
-
```
Comment (AbhishekA1509) [inline src/Common/AppStatus/utils.ts:2]: Try to have a blank linke between component and import statements for readability purpose

File: src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+const InfoIconTippy = ({
+    heading,
+    infoText,
+    iconClass = 'fcv-5',
+    documentationLink,
+    documentationLinkText,
+    additionalContent,
+    className = 'icon-dim-16',
+    placement = 'bottom',
+    dataTestid,
```
Comment (AbhishekA1509) [inline src/Shared/Components/InfoIconTippy/InfoIconTippy.tsx:15]: Can we add a default test id

**Review by AbhishekA1509 — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**